### DISCUSSION TITLE
Can I get app.storage.browser['id'] in the cpu_bound task?

### DISCUSSION THREAD

**Discussion by KimigaiiWuyi** [zauberzeug/nicegui#2903] [Q&A]
### Question

I have a CPU-intensive task that contains the part `app.storage.browser['id']` that gets the ID, 
and when I try to run it, it reports the error
```shell
RuntimeError: app.storage.browser can only be used with page builder functions (https://nicegui.io/documentation/page)
```

**Comment by KimigaiiWuyi** [comment]
If that's not possible, do I have the option of getting `app.storage.browser['id']` in advance and passing it to the function, but that would undoubtedly break the encapsulation of the function, is there a simpler way?

**Comment by vickorian** [comment]
@KimigaiiWuyi how are you using it in your task? Can you show an example of how you're calling it?

**Comment by rodja** [comment]
> do I have the option of getting app.storage.browser['id'] in advance and passing it to the function, 

Sure!

```py
await run.cpu_bound(my_heavy_computation, app.storage.browser['id'])
```

> but that would undoubtedly break the encapsulation of the function, is there a simpler way?

Why does it break encapsulation? The id is surely something external which should be ok to pass into the function. As written in the docstring and noted in https://github.com/zauberzeug/nicegui/wiki/FAQs#why-is-my-long-running-function-blocking-ui-updates, there is no way around transferring the whole state from one process to the other.

**Reply by KimigaiiWuyi** [reply]
I have a global task object that uses an id to identify the user, I execute the task in a cpu_bound task and reply to the task in another task, as a producer-consumer model, both tasks use the id to get the current task status, it seems that the code would be a bit cumbersome if each task needs to pass the id.

Maybe it's an xy problem, I'll try to pass in the id to see if it's intuitive.
